src/components/tasks/card/index.tsx

```diff
@@ -121,7 +132,51 @@ const Card: FC<Props> = ({
       </span>
       )
   }
- 
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: SELF_URL });
+        const { data } = await requestPromise;
+        const userRoles = {
+          adminUser: data.roles?.admin,
+          superUser: data.roles?.super_user,
+        };
+        const { adminUser, superUser } = userRoles;
+        setIsUserAuthorized(!!adminUser || !!superUser);
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
+    return (() => {
+      setIsUserAuthorized(false);
+    });
+  }, []);
+
+  function handleKeyDown(event: any): void {
+    const ALT_KEY = 18;
+    if (event.keyCode === ALT_KEY) {
+      setShowEditButton(true);
+    }
+  }
+
+  const debouncedHandler = useDebounce(handleKeyDown, 300);
+
+  useEffect(() => {
+
+    document?.addEventListener('keydown', debouncedHandler);
```

Comment:
do we require `?` operator here.

Reply:
What would you suggest to handle the scenario when the document is undefined @heyrandhir

Reply:
yeah Right. that would be needed. if the document is undefined. I assumed it would never be undefined. My bad. Nicely handled @Maheima .